src/pages/settings/Security/TwoFactorAuth/PageWrapper.tsx

```diff
@@ -0,0 +1,100 @@
+import React, {useMemo} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import type {ValueOf} from 'type-fest';
+import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
+import DelegateNoAccessWrapper from '@components/DelegateNoAccessWrapper';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import {quitAndNavigateBack} from '@libs/actions/TwoFactorAuthActions';
+import CONST from '@src/CONST';
+import type {StepCounterParams} from '@src/languages/params';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
+
+type PageWrapperProps = ChildrenProps & {
+    /** Name of the step */
+    stepName: ValueOf<typeof CONST.TWO_FACTOR_AUTH_STEPS>;
+
+    /** Title of the Header */
+    title: string;
+
+    /** Data to display a step counter in the header */
+    stepCounter?: StepCounterParams;
+
+    /** Method to trigger when pressing back button of the header */
+    onBackButtonPress?: () => void;
+
+    /** Flag to indicate if the keyboard avoiding view should be enabled */
+    shouldEnableKeyboardAvoidingView?: boolean;
+};
+
+function PageWrapper({stepName, title, stepCounter, onBackButtonPress, shouldEnableKeyboardAvoidingView = true, children}: PageWrapperProps) {
```

Comment:
Why do we need this new wrapper?

Reply:
This is a wrapper only for 2fa pages, without it there would be a lot of duplicated code